### PR TITLE
New semantic slots (ref: Events WebPart)

### PR DESCRIPTION
**Pull request by bennettclark**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adds 5 new semantic slots: actionLink, actionLinkHovered, cardBackground, bodyStandoutBackground, variantBorder

### CODE REVIEW

File: packages/styling/src/styles/theme.ts [hunk 1]
```diff
@@ -144,20 +144,24 @@ export function createTheme(theme: IPartialTheme, depComments: boolean = false):
 // We'll use these as fallbacks for semantic slots that the passed in theme did not define.
 function _makeSemanticColorsFromPalette(p: IPalette, isInverted: boolean, depComments: boolean): ISemanticColors {
   let toReturn: ISemanticColors = {
+    actionLink: p.neutralPrimary,
```
Comment (phkuo) [inline packages/styling/src/styles/theme.ts]: put these down with the link slots (in this file and elsewhere)

File: packages/styling/src/interfaces/ISemanticColors.ts [hunk 2]
```diff
@@ -51,6 +51,16 @@ export interface ISemanticColors {
 
   //// Base slots
 
+  /**
```
Comment (phkuo) [inline packages/styling/src/interfaces/ISemanticColors.ts]: We wanted to add a bodyStandoutBackground too right?

Comment (bennettclark) [reply packages/styling/src/interfaces/ISemanticColors.ts]: Per the designer: "for standouts, we have bodyFrameBackground (hub nav in header) and standoutCardBackground (news hub/top news, some cards)" - not bodyStandoutBackground.

The designer would prefer to add new semantic slots as each redline is finalized. The two slots mentioned above (bodyFrameBackground and standoutCardBackground) are not applicable to the events webpart, so they are slated to be added later.

File: packages/styling/src/styles/theme.ts (hunk 1 above)
Comment (bennettclark) [reply packages/styling/src/styles/theme.ts]: Moved actionlink slots to be grouped with link slots. Thanks!

**Comment by phkuo** [comment]
@bennettclark I thought for cards, we were adding bodyStandoutBackground, and then the cards would conditionally switch between using bodyBackground and bodyStandoutBackground. And same with the standout card with headlines.
This is because the card itself determines its variant theme based on its context's variant theme, therefore having its own semantic slot does nothing, because that semantic slot will always be the value at that self-chosen variant, instead of being a value based on the parent's variant.
e.g. Dark theme no variant: cards use neutral variant. Dark theme neutral variant: cards are STILL using neutral variant, except with the standout background. In both cases the cards are using the same variant therefore any semantic slots shared between them are the same, therefore you need another layer of logic to be able to switch between them.

**Comment by bennettclark** [comment]
@phkuo I've added bodyStandoutBackground with default color of white. Let me know if there's anything else you'd like on this!

File: packages/styling/src/interfaces/ISemanticColors.ts [hunk 3]
```diff
@@ -56,6 +56,11 @@ export interface ISemanticColors {
    */
   bodyBackground: string;
 
+  /**
+   * The standout color for backgrounds.
```
Comment (phkuo) [inline packages/styling/src/interfaces/ISemanticColors.ts]: Describe briefly that standout is highlighted content, and that it should be using the neutral variant when on no emphasis, and that it's a shade darker the other emphases.

File: packages/styling/src/interfaces/ISemanticColors.ts [hunk 4]
```diff
@@ -92,6 +97,21 @@ export interface ISemanticColors {
    */
   bodyDivider: string;
 
+  /**
+   * Background color for "cards" (elements with a background that sit on top of another background).
+   */
+  cardBackground: string;
```
Comment (phkuo) [inline packages/styling/src/interfaces/ISemanticColors.ts]: remove this slot

File: packages/styling/src/styles/theme.ts [hunk 5]
```diff
@@ -145,19 +145,22 @@ export function createTheme(theme: IPartialTheme, depComments: boolean = false):
 function _makeSemanticColorsFromPalette(p: IPalette, isInverted: boolean, depComments: boolean): ISemanticColors {
   let toReturn: ISemanticColors = {
     bodyBackground: p.white,
+    bodyStandoutBackground: p.white,
```
Comment (phkuo) [inline packages/styling/src/styles/theme.ts]: you'll also want to add the new redlines in variants:
https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/variants/src/variants.ts
(might as well do it all at once)

Comment (bennettclark) [reply packages/styling/src/styles/theme.ts]: @phkuo what do you mean by "add the new redlines"?

Comment (bennettclark) [reply packages/styling/src/styles/theme.ts]: @phkuo Oh, do you mean add bodyStandoutBackground:neutralLight into getNeutralVariant() and getStrongVariant() inside the partialSemantic const? I can do that. Fingers crossed for no conflicts with the other PR about variants.ts.

Comment (bennettclark) [reply packages/styling/src/styles/theme.ts]: Ok, added. Let me know if there's anything further needed on that.

File: packages/variants/src/variants.ts [hunk 6]
```diff
@@ -178,6 +179,7 @@ export function getStrongVariant(theme: IPartialTheme): ITheme {
 
   const partialSemantic: Partial<ISemanticColors> = {
     bodyBackground: p.themePrimary,
+    bodyStandoutBackground: p.neutralLight,
```
[the remark this answers is not in the document]
Comment (bennettclark) [reply packages/variants/src/variants.ts]: @phkuo Oh hmm. So you said to "add the redlines" so I figured the value for the slot must come from the redlines. How can I find what p.themeSomething this should have?

[the remark this answers is not in the document]
Comment (bennettclark) [reply packages/variants/src/variants.ts]: Updated! Thanks!

File: packages/variants/src/variants.ts [hunk 7]
```diff
@@ -59,6 +59,7 @@ export function getNeutralVariant(theme: IPartialTheme): ITheme {
 
   const partialSemantic: Partial<ISemanticColors> = {
     bodyBackground: p.neutralLighter,
+    bodyStandoutBackground: p.themeDarkAlt,
```
Comment (phkuo) [inline packages/variants/src/variants.ts]: this should be one step up from neutralLighter, which is neutralLight
also need this slot for getSoftVariant
if the variantBorder values are finalized, i'd add those here too

File: packages/styling/src/interfaces/ISemanticColors.ts [hunk 8]
```diff
@@ -56,6 +56,13 @@ export interface ISemanticColors {
    */
   bodyBackground: string;
 
+  /**
+   * The standout color for highlighted content backgrounds.
+   * This should use the neutral variant when on no emphasis.
```
Comment (phkuo) [inline packages/styling/src/interfaces/ISemanticColors.ts]: More accurately,
"For highlighted content when there is no emphasis, use the neutral variant instead."

File: packages/styling/src/interfaces/ISemanticColors.ts [hunk 9]
```diff
@@ -56,6 +56,13 @@ export interface ISemanticColors {
...
+   * This should be a shade darker than the other emphases.
```
Comment (phkuo) [inline packages/styling/src/interfaces/ISemanticColors.ts]: "shade darker than bodyBackground"
and something about it being lighter in inverted themes

**Pull request merged**